peekingduck/pipeline/nodes/model/yolact_edgev1/yolact_edge_files/model.py

```diff
@@ -0,0 +1,653 @@
+            prior_data = []
+            for j, i in product(range(conv_h), range(conv_w)):
+                # +0.5 because priors are in center-size notation
+                center_x = (i + 0.5) / conv_w
+                center_y = (j + 0.5) / conv_h
+                for scale, aspect_ratios in zip(self.scales, self.aspect_ratios):  # type: ignore
+                    for aspect_ratio in aspect_ratios:
+                        aspect_ratio = sqrt(aspect_ratio)
+                        width = scale * aspect_ratio / 550
+                        height = width
+                        prior_data += [center_x, center_y, width, height]
+            self.priors = torch.Tensor(prior_data).view(-1, 4)
+            self.last_conv_size = (conv_w, conv_h)
+        return self.priors
+
+
+class FPNPhase1(ScriptModuleWrapper):
+    """First phase of the feature pyramid network"""
+
+    __constants__ = ["interpolation_mode"]
+
+    def __init__(self, in_channels: List[int]) -> None:
+        super().__init__()
+        self.src_channels = in_channels
+        self.lat_layers = nn.ModuleList(
+            [
+                nn.Conv2d(x, FPN_NUM_FEATURES, kernel_size=1)
+                for x in reversed(in_channels)
+            ]
+        )
+        self.interpolation_mode = "bilinear"
+
+    def forward(  # pylint: disable=too-many-arguments, too-many-locals
+        self,
+        x_1: Optional[torch.Tensor] = None,
+        x_2: Optional[torch.Tensor] = None,
+        x_3: Optional[torch.Tensor] = None,
+        x_4: Optional[torch.Tensor] = None,
+        x_5: Optional[torch.Tensor] = None,
+        x_6: Optional[torch.Tensor] = None,
+        x_7: Optional[torch.Tensor] = None,
+    ) -> List[Tensor]:
+        """
+        Args:
+            - convouts (list): A list of convouts for the corresponding layers
+                in in_channels.
+        Returns:
+            - A list of FPN convouts in the same order as x with extra downsample
+                layers if requested.
+        """
+        convouts_ = [x_1, x_2, x_3, x_4, x_5, x_6, x_7]
+        convouts = []
+
+        for count, _ in enumerate(convouts_):
+            if convouts_[count] is not None:
+                convouts.append(convouts_[count])
+
+        out = []
+        lat_feats = []
+        x_0 = torch.zeros(1)
+
+        for i in range(len(convouts)):
+            out.append(x_0)
+            lat_feats.append(x_0)
+
+        count = len(convouts)
+        for lat_layer in self.lat_layers:
+            count -= 1
+            if count < len(convouts) - 1:
+                _, _, height, weight = convouts[count].size()  # type: ignore
+                x_0 = F.interpolate(
+                    x_0,
+                    size=(height, weight),
+                    mode=self.interpolation_mode,
+                    align_corners=False,
+                )
+            lat_iter = lat_layer(convouts[count])
+            lat_feats[count] = lat_iter
+            x_0 = x_0 + lat_iter
```

Comment:
To refactor as `x_0 += lat_iter`

Reply:
Tech note: in this context, `x_0 = x_0 + lat_iter` will have a different effect from `x_0 += lat_iter`, so cannot replace the former with the latter.